src/test/java/org/broadinstitute/hellbender/utils/read/GATKReadAdaptersUnitTest.java

```diff
@@ -1394,4 +1394,59 @@ private int parseSAMStringFlags(String samString) {
         Assert.assertTrue(m.find());
         return Integer.parseInt(m.group(2)); // use the second capture group in the reg ex above since we don't want the tab
     }
+
+    @DataProvider(name="copyData")
+    public Object[][] getCopyData() {
+        List<Object[]> testCases = new ArrayList<>();
+
+        testCases.add(new Object[]{basicReadBackedBySam()});
+        testCases.add(new Object[]{basicReadBackedByGoogle()});
+
+        return testCases.toArray(new Object[][]{});
+    }
+
+    @Test(dataProvider="copyData")
+    public void testShallowCopy(final GATKRead read) {
+        Assert.assertTrue(read.equals(read.copy()));
+
+        read.setIsReverseStrand(false);
+        GATKRead shallowCopy = read.copy();
+        Assert.assertTrue(read.isReverseStrand() == shallowCopy.isReverseStrand());
+        read.setIsReverseStrand(true);
+        Assert.assertFalse(read.isReverseStrand() == shallowCopy.isReverseStrand());
+        Assert.assertFalse(read.equals(shallowCopy));
+    }
+
+    @Test(dataProvider="copyData")
+    public void testDeepCopy(final GATKRead read) {
+        Assert.assertTrue(read.equals(read.deepCopy()));
+
+        GATKRead deepCopy;
+
+        // mutate a few things that are deeply nested in one or other of the various backing
+        // implementations to make sure we've copied deeply; i.e. for GoogleRead backed reads,
+        // reverseStrand is nested as LinearAlignment->Position->reverseStrand
+        read.setIsReverseStrand(false);
+        deepCopy = read.deepCopy();
+        Assert.assertTrue(read.isReverseStrand() == deepCopy.isReverseStrand());
+        read.setIsReverseStrand(true);
+        Assert.assertFalse(read.isReverseStrand() == deepCopy.isReverseStrand());
+        Assert.assertFalse(read.equals(deepCopy));
+
+        // attributes
+        deepCopy = read.deepCopy();
```

Comment:
these reassignments suggest splitting this test method into 3 methods `testDeepCopyStrand`, `testDeepCopyAttributes` and `testDeepCopyPosition`